.clang-tidy

```diff
@@ -16,6 +16,7 @@ Checks: >
     -cppcoreguidelines-non-private-member-variables-in-classes,
     -cppcoreguidelines-macro-usage,
     -fuchsia-overloaded-operator,-fuchsia-default-arguments*,
+    -cppcoreguidelines-pro-type-reinterpret-cast,
```

Comment:
there are too many `reinterpret_cast` operations due to our low-level implementation manner, so disable the warning directly.